src/uct/tcp/tcp_iface.c

```diff
@@ -122,25 +129,30 @@ static ucs_status_t uct_tcp_iface_query(uct_iface_h tl_iface, uct_iface_attr_t *
     attr->cap.am.max_short = am_buf_size;
     attr->cap.am.max_bcopy = am_buf_size;
 
-    if (iface->config.zcopy.max_iov > UCT_TCP_EP_AM_ZCOPY_SERVICE_IOV_COUNT) {
+    if (iface->config.zcopy.max_iov > UCT_TCP_EP_ZCOPY_SERVICE_IOV_COUNT) {
+        /* AM */
         attr->cap.am.max_iov          = iface->config.zcopy.max_iov -
-                                        UCT_TCP_EP_AM_ZCOPY_SERVICE_IOV_COUNT;
+                                        UCT_TCP_EP_ZCOPY_SERVICE_IOV_COUNT;
         attr->cap.am.max_zcopy        = iface->config.rx_seg_size -
                                         sizeof(uct_tcp_am_hdr_t);
         attr->cap.am.max_hdr          = iface->config.zcopy.max_hdr;
         attr->cap.am.opt_zcopy_align  = 512;
+        attr->cap.am.align_mtu        = mtu;
         attr->cap.flags              |= UCT_IFACE_FLAG_AM_ZCOPY;
-    }
 
-    status = uct_tcp_netif_caps(iface->if_name, &attr->latency.overhead,
-                                &attr->bandwidth.shared, &attr->cap.am.align_mtu);
-    attr->bandwidth.dedicated = 0;
-    if (status != UCS_OK) {
-        return status;
+        /* PUT */
+        attr->cap.put.max_iov          = iface->config.zcopy.max_iov -
+                                         UCT_TCP_EP_ZCOPY_SERVICE_IOV_COUNT;
+        attr->cap.put.max_zcopy        = UCT_TCP_EP_PUT_ZCOPY_MAX -
+                                         UCT_TCP_EP_PUT_SERVICE_LENGTH;
+        attr->cap.put.opt_zcopy_align  = 512;
+        attr->cap.put.align_mtu        = mtu;
```

Comment:
this is not really needed for TCP

Reply:
should we remove it for AM and PUT? or PUT only?

Reply:
for both